uplink/piecestore/upload.go

```diff
@@ -67,7 +67,14 @@ func (client *Client) Upload(ctx context.Context, limit *pb.OrderLimit, piecePri
 	})
 	if err != nil {
 		_, closeErr := stream.CloseAndRecv()
-		return nil, ErrProtocol.Wrap(errs.Combine(err, closeErr))
+		switch {
+		case err != io.EOF && closeErr != nil:
+			err = ErrProtocol.Wrap(errs.Combine(err, closeErr))
+		case closeErr != nil:
+			err = ErrProtocol.Wrap(closeErr)
```

Comment:
this will make us lose the original `err`. Is that okay or should we combine them?

Reply:
we don't need to combine with `io.EOF` error, which is when this switch branch happens.

Reply:
oh that makes sense

Reply:
Do we need to close stream if there is no error?

Reply:
It's closed if there is an error.
`if err != nil` line 70 (above)